Source/FlowEditor/Private/Graph/FlowGraphEditor.cpp

```diff
@@ -609,83 +628,179 @@ void SFlowGraphEditor::PasteNodes()
+
+			PasteNode->SnapToGrid(16);
+
+			// Give new node a different Guid from the old one
+			PasteNode->CreateNewGuid();
 
-		// Give new node a different Guid from the old one
-		Node->CreateNewGuid();
+			if (UFlowNode* FlowNode = Cast<UFlowNode>(PasteFlowGraphNode->GetFlowNodeBase()))
+			{
+				// Only full FlowNodes are registered with the Asset
+				// (for now?  perhaps we register AddOns in the future?)
+				FlowAsset->RegisterNode(PasteNode->NodeGuid, FlowNode);
+			}
+		}
 
-		if (const UFlowGraphNode* FlowGraphNode = Cast<UFlowGraphNode>(Node))
+		if (PasteFlowGraphNode)
 		{
-			FlowAsset->RegisterNode(Node->NodeGuid, FlowGraphNode->GetFlowNode());
+			PasteFlowGraphNode->RemoveAllSubNodes();
 		}
+	}
 
-		// Select the newly pasted stuff
-		SetNodeSelection(Node, true);
+	for (TSet<UEdGraphNode*>::TIterator It(PastedNodes); It; ++It)
+	{
+		UFlowGraphNode* PasteNode = Cast<UFlowGraphNode>(*It);
+		if (PasteNode && PasteNode->IsSubNode())
+		{
+			PasteNode->NodePosX = 0;
+			PasteNode->NodePosY = 0;
 
-		Node->NodePosX = (Node->NodePosX - AvgNodePosition.X) + Location.X;
-		Node->NodePosY = (Node->NodePosY - AvgNodePosition.Y) + Location.Y;
+			// remove subnode from graph, it will be referenced from parent node
+			PasteNode->DestroyNode();
 
-		Node->SnapToGrid(SNodePanel::GetSnapGridSize());
+			if (PasteNode->CopySubNodeParentIndex == INDEX_NONE)
+			{
+				// INDEX_NONE parent index indicates we should set the parent to the SelectedParent
+				
+				if (SelectedParent)
+				{
+					SelectedParent->AddSubNode(PasteNode, FlowGraph);
+				}
+			}
+			else if (UFlowGraphNode* PastedParentNode = EdNodeCopyIndexMap.FindRef(PasteNode->CopySubNodeParentIndex))
+			{
+				PastedParentNode->AddSubNode(PasteNode, FlowGraph);
+			}
+		}
 	}
 
-	// Force new pasted FlowNodes to have same connections as graph nodes
-	FlowAsset->HarvestNodeConnections();
+	if (FlowGraph)
+	{
+		FlowGraph->UpdateClassData();
+		FlowGraph->OnNodesPasted(TextToImport);
+		FlowGraph->UnlockUpdates();
+	}
 
 	// Update UI
 	NotifyGraphChanged();
 
-	FlowAsset->PostEditChange();
-	FlowAsset->MarkPackageDirty();
+	UObject* GraphOwner = FlowGraph->GetOuter();
+	if (GraphOwner)
+	{
+		GraphOwner->PostEditChange();
+		GraphOwner->MarkPackageDirty();
+	}
 }
 
 bool SFlowGraphEditor::CanPasteNodes() const
 {
-	if (CanEdit() && IsTabFocused())
+	if (!CanEdit() || !IsTabFocused())
 	{
-		FString ClipboardContent;
-		FPlatformApplicationMisc::ClipboardPaste(ClipboardContent);
+		return false;
+	}
+
+	FString ClipboardContent;
+	FPlatformApplicationMisc::ClipboardPaste(ClipboardContent);
+
+	const bool bIsPastePossible = FEdGraphUtilities::CanImportNodesFromText(FlowAsset->GetGraph(), ClipboardContent);
 
-		return FEdGraphUtilities::CanImportNodesFromText(FlowAsset->GetGraph(), ClipboardContent);
+	if (!bIsPastePossible)
+	{
+		return false;
 	}
 
-	return false;
+	// TODO (gtaylor) Need to confirm the nodes are allowed to be pasted on the selected node(s)
```

Comment:
This is a known problem.  You can paste AddOns where they are not normally allowed, at this time.

We'll be fixing that later this month~.